services/src/main/java/org/keycloak/services/clientpolicy/executor/DPoPBindEnforcerExecutorFactory.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.services.clientpolicy.executor;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.keycloak.Config.Scope;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.provider.ProviderConfigProperty;
+
+public class DPoPBindEnforcerExecutorFactory  implements ClientPolicyExecutorProviderFactory {
```

Comment:
Should this maybe implement `EnvironmentDependentFactory` and have this supported just when DPoP feature is enabled? This means that no check is needed in `DPoPBindEnforcerExecutor` itself.

Reply:
Yes, I fill revise it as you suggested.

Reply:
Is it possible to address this point?

Reply:
Sorry, I had forgot it. I added `EnvironmentDependentFactory ` to `DPoPBindEnforcerExecutorFactory`.